pandas/tests/test_algos.py

```diff
@@ -436,13 +436,13 @@ def test_factorize_na_sentinel(self, sort, na_sentinel, data, uniques):
         [
             (
                 ["a", None, "b", "a"],
-                np.array([0, 2, 1, 0], dtype=np.dtype("intp")),
-                np.array(["a", "b", np.nan], dtype=object),
+                np.array([0, 1, 2, 0], dtype=np.dtype("intp")),
+                np.array(["a", None, "b"], dtype=object),
```

Comment:
The change from np.nan to None here was an unintended side-effect; is this undesirable?

Reply:
IMO I think this could be called a bug fix since now a user can preserve None in their input data when using `pd.factorize`. Probably good to add a whatsnew note for this chage.

Reply:
Thanks! I completely forgot about this open question. Will do.